components/aws/sagemaker/tests/integration_tests/utils/ack_utils.py

```diff
@@ -10,40 +10,52 @@ def k8s_client():
 def _get_resource(k8s_client, job_name, plural):
     """Get the custom resource detail similar to: kubectl describe <resource> JOB_NAME -n NAMESPACE.
     Returns:
-        None or object: None if the resource doesnt exist in server, otherwise the
+        None or object: None if the resource doesn't exist in server or there is an error, otherwise the
             custom object.
     """
     _api = client.CustomObjectsApi(k8s_client)
     namespace = os.environ.get("NAMESPACE")
-    job_description = _api.get_namespaced_custom_object(
-        "sagemaker.services.k8s.aws",
-        "v1alpha1",
-        namespace.lower(),
-        plural,
-        job_name.lower(),
-    )
+    try:
+        job_description = _api.get_namespaced_custom_object(
+            "sagemaker.services.k8s.aws",
+            "v1alpha1",
+            namespace.lower(),
+            plural,
+            job_name.lower(),
+        )
+    except Exception as e:
+        print(f"Exception occurred while getting resource {job_name}: {e}")
+        return None
     return job_description
 
 
-def _delete_resource(k8s_client, job_name, plural):
+def _delete_resource(k8s_client, job_name, plural, wait_periods=10, period_length=20):
     """Delete the custom resource
     Returns:
-        None or object: None if the resource doesnt exist in server, otherwise the
-            custom object.
+        True or False: True if the resource is deleted, False if the resource deletion times out
     """
     _api = client.CustomObjectsApi(k8s_client)
     namespace = os.environ.get("NAMESPACE")
-    try:
-        _api.delete_namespaced_custom_object(
-            "sagemaker.services.k8s.aws",
-            "v1alpha1",
-            namespace.lower(),
-            plural,
-            job_name.lower(),
-        )
-    except:
-        return False
-    return True
+    for i in range(wait_periods):
+        if _get_resource(k8s_client, job_name, plural) is None:
+            print(f"Resource {job_name} deleted successfully.")
+            return True
+        else:
+            try:
+                _api.delete_namespaced_custom_object(
```

Comment:
delete once and then just check for resource clean up, deleting multiple times can sometimes produce incorrect result where resource might get cleaned up right after get call on line 40